gmso/tests/test_units.py

```diff
@@ -0,0 +1,243 @@
+        usys = LAMMPS_UnitSystems("electron")
+        parametersList = list(
+            map(
+                lambda x: 1 * u.Unit(x, registry=usys.reg),
+                [
+                    "nm",
+                    "kJ",
+                    "kJ/mol",
+                    "K",
+                    "degree/angstrom",
+                    "elementary_charge/mm",
+                    "dimensionless",
+                    "kg*m**2/s**2",
+                    "coulomb",
+                    "kcal/nm**2",
+                    "K/nm",
+                ],
+            )
+        )
+
+        output_dimensionsList = [
+            "length",
+            "energy",
+            "energy",
+            "temperature",
+            "angle/length",
+            "charge/length",
+            "dimensionless",
+            "energy",
+            "charge",
+            "energy/length**2",
+            "energy/length",
+        ]
+        for parameter, dim in zip(parametersList, output_dimensionsList):
+            if str(parameter.units) == "K/nm":
+                thermalize = True
+            else:
+                thermalize = False
+            dimsStr = str(
+                usys._get_output_dimensions(
+                    parameter.units.dimensions, thermalize
+                )
+            )
+            remove_parStr = dimsStr.translate({ord(i): None for i in "()"})
+            assert remove_parStr == str(dim)
+
+    def test_convert_parameters(self, typed_ethane, real_usys):
+        parameter = typed_ethane.sites[0].atom_type.parameters["epsilon"]
+        assert real_usys.convert_parameter(parameter) == "0.066"
+        real_usys.usystem["energy"] = u.kJ / u.mol
+        assert (
+            real_usys.convert_parameter(parameter, n_decimals=6) == "0.276144"
+        )
+        usys = LAMMPS_UnitSystems("real")
+        parameter = typed_ethane.bonds[0].bond_type.parameters["k"]
+        assert usys.convert_parameter(parameter, n_decimals=0) == "680"
+        parameter = typed_ethane.angles[0].angle_type.parameters["theta_eq"]
+        assert usys.convert_parameter(parameter, name="theta_eq") == "110.700"
+        parameter = typed_ethane.dihedrals[0].dihedral_type.parameters["c0"]
+        assert usys.convert_parameter(parameter) == "0.150"
+
+    def test_get_parameter_dimension(self):
+        from gmso.utils.units import get_parameter_dimension
+
+        assert (
+            get_parameter_dimension(1 * u.kJ / u.mol / u.nm, "(energy)")
+            == u.kJ / u.mol
+        )
+        assert get_parameter_dimension(1 * u.kJ / u.nm, "(length)") == u.nm
+        assert (
+            get_parameter_dimension(1 * u.kJ / u.mol / u.nm, "(length)") == u.nm
+        )
+
+    def test_convert_to_unit_system(self):
+        # TODO: discuss if we would want a function to convert a whole
+        # topology at once.
+        # convert a whole topology to a specific unit system
+        pass
+
+    def test_generate_unit_styles(self):
```

Comment:
Would be useful to have too, or at least show what unit systems we are currently support.